### DISCUSSION TITLE
Can you use `useField` outside a form?

### DISCUSSION THREAD

**Discussion by mattbeiswenger** [airjp73/rvf#374] [Q&A] [answered]
The [documentation](https://www.remix-validated-form.io/integrate-your-components) states the following:

 > `useField` consumes the context provided by `ValidatedForm`, so these features only work if you use it in a `ValidatedForm`. However, it is still safe to use your input outside of a `ValidatedForm`, it just won't be able to take advantage of the features.

However, when using `useField` outside of a `ValidatedForm` you get the error:

```
Unable to determine form for useField. Please use it inside a ValidatedForm or pass a 'formId'.
```

Is the documentation just incorrect or is there a way to use `useField` outside a `ValidatedForm`? My use case is that I would like to slowly convert our forms to using validation, however, when I change low-level components like input elements to use `useField`, validation breaks. Currently, for the migration process, I'm resorting to having two copies of each component (i.e. `SingleSelect` and `SingleSelectNoValidations`). While this works fine, it would obviously be nicer to keep one component and have it work with or without a wrapping `ValidatedForm` but just foregoing validation features when it's not wrapped.

**Comment by airjp73** [comment, marked as answer]
This line sounds outdated, it's wrong or misleading. It might be referring to the [use outside forms feature] (https://www.remix-validated-form.io/use-outside-forms), but that still requires there to be a form somewhere.

I think the approach of having an RVF specific component that wraps the non-validated one is the right approach.

FWIW, v6 doesn't require these wrapper components to function properly (but you can still make them if you want). They can be wired up directly in your form similarly to react-hook-form's register function.
